sycl/include/sycl/ext/oneapi/experimental/builtins.hpp

```diff
@@ -108,10 +108,41 @@ inline __SYCL_ALWAYS_INLINE
 #endif
 }
 
+// These marray math function implementations use vectorizations of
+// size two as a simple general optimization. A more complex implementation
+// using larger vectorizations for large marray sizes is possible; however more
+// testing is required in order to ascertain the performance implications for
+// all backends.
+template <typename T, size_t N>
+inline __SYCL_ALWAYS_INLINE std::enable_if_t<std::is_same<T, half>::value ||
+                                                 std::is_same<T, float>::value,
+                                             sycl::marray<T, N>>
+tanh(sycl::marray<T, N> x) __NOEXC {
+  sycl::marray<T, N> res;
+#if defined(__SYCL_DEVICE_ONLY__) && defined(__NVPTX__)
+#define FUNC_VEC native::tanh
```

Comment:
Could you please add a comment on why NVPTX needs to use native here and for exp2?

Reply:
Sure I can add an appropriate comment in the code

For reviewers reference: `sycl::tanh` and `sycl::exp2` are already supported for nvptx in the sycl/builtins.hpp header. The `sycl::native::tanh` does not exist in the SYCL spec and so it is added here: `sycl::ext::oneapi::experimental::tanh`. `sycl::native::exp2` does exist in the SYCL spec, however not for the half type, so the half case for exp2 is added here also.
The reason why NVPTX wants to use native implementations tanh and exp2 is, i guess, because they are frequently used functions.
The reason why we want to add them as an extension is 
1) The above reason
and
2) We want to support all ptx 7.0 instructions.

Reply:
I'll also add a note saying that only nvptx backend supports `native::tanh`/`native::ex2`(for half) currently and other backends are set to fall back on `sycl::tanh` here.